lib/internal/process/promises.js

```diff
@@ -20,6 +23,15 @@ const pendingUnhandledRejections = [];
 const asyncHandledRejections = [];
 let lastPromiseId = 0;
 
+const states = {
+  none: 0,
```

Comment:
minor non-blocking nit... I'd prefer if these were defined as constants

```js
const kStateNone = 0;
const kStateWarn = 1;
const kStateStrict = 2;
const kStateDefault = 3;
```

And would prefer a simple switch rather than an object lookup. For super small sets like this the switch should be faster

Reply:
I'll open a follow-up PR for the two nits. I would like to land this as is, otherwise other's have to look at this again.

Reply:
The switch case is only executed once, so it's not about performance.